src/components/Radio/RadioGroup.svelte

```diff
@@ -1,77 +1,64 @@
-	import { createEventDispatcher } from 'svelte'
-	import { get_current_component } from 'svelte/internal'
-
-	import { forwardEventsBuilder } from '@ubeac/svelte/directives'
-	import { nanoid } from 'nanoid'
-
-	import { FormRadio } from '$lib/components'
-	import type { Colors } from '$lib/types'
-	import { classname, createOptions } from '$lib/utils'
-
-	import Radio from './Radio.svelte'
+	import { El, Radio } from '$lib/components'
+	import type { RadioGroupProps } from '$lib/components'
 
+	type $$Props = RadioGroupProps
 	/**
-	 * Set Color of RadioGroup
+	 * Set color for the selected Radio's background color
 	 */
-	export let color: Colors = 'default'
+	export let color: $$Props['color'] = undefined
 
 	/**
-	 * Forwards All native events.
+	 * Set Css Prefix for the RadioGroup
 	 */
-	export let forwardEvents = forwardEventsBuilder(get_current_component())
+	export let cssPrefix: $$Props['cssPrefix'] = 'radio-group'
 
 	/**
-	 * Set View of RadioGroup
+	 * Set inline to sohw inline Radios
 	 */
-	export let inline: boolean = false
+	export let inline: $$Props['inline'] = undefined
 
 	/**
-	 * Select items
+	 * Array of items to be bound to the Radios
 	 */
-	export let items: any = []
+	export let items: $$Props['items'] = undefined
 
 	/**
-	 * Set the field that selected options are bound to
+	 * Set reverse to show reversed Radios
 	 */
-	export let key: string = 'value'
+	export let reverse: $$Props['reverse'] = undefined
 
 	/**
-	 * Set name of RadioGroup
+	 * The value of RadioGroup
 	 */
-	export let name: string = 'radio-group-' + nanoid(5)
-
-	/**
-	 * Set the field that should be used as each option label
-	 */
-	export let text: string = 'text'
-
-	/**
-	 * Value that selected radio text is bound to
-	 */
-	export let value: number | string | undefined = undefined
-
-	const dispatch = createEventDispatcher()
-
-	function onChange(checked: any, option: any) {
-		if (checked) value = getKey(option)
-		dispatch('changed', value)
+	export let value: $$Props['value'] = undefined
+
+	let element: any = null
+	let radioProps: any = {}
+
+	$: {
+		radioProps = {
+			inline,
+			reverse,
+			name: element?.id,
+			color,
+		}
 	}
 
-	$: ({ options, getKey, getText, isSelected } = createOptions({ items, key, text }))
-
-	$: classes = classname('radio-group', {}, $$props.class)
+	const onChange = (event: any) => {
+		const selectedIndex = event.target.value
+		value = items && selectedIndex ? items[selectedIndex] : undefined
+	}
 </script>
 
-<div class={classes} role="radiogroup">
-	{#each $options as option}
-		<Radio
-			{forwardEvents}
-			{name}
-			{color}
-			{inline}
-			value={isSelected(option, value)}
-			text={getText(option)}
-			on:changed={({ detail }) => onChange(detail, option)} />
-	{/each}
-</div>
+<El {cssPrefix} bind:element {...$$restProps}>
+	{#if items}
+		{#each items as item, index}
+			<Radio {...radioProps} value={index} checked={value === item} on:change={onChange}>
+				<slot {index} {item}>{item}</slot>
```

Comment:
This solution is better than context and stores (I think you tried in previous commits).
it would be great if we could use `createOptions` or something similar to render Radios based on 'key' and 'text' props. 

```
<script>

</script>
<RadioGroup items=

```